kdc/kerberos5.c

```diff
@@ -38,6 +38,44 @@
 #undef __attribute__
 #define __attribute__(X)
 
+struct kdc_pa_auth_state {
+	int auth_status;
+	const char *auth_details;
+	void *free_ptr;
+};
+
+static krb5_error_code
+_kdc_audit_auth_status(astgs_request_t r,
+		       int auth_status,
+		       const char *auth_details,
+		       const char *pa_type)
+{
+    krb5_context context = r->context;
+    krb5_kdc_configuration *config = r->config;
+    struct HDB *hdb = NULL;
+    int ret = 0;
+
+    if (r->clientdb != NULL) {
+	hdb = r->clientdb;
+    } else if (config->db[0]) {
+	hdb = config->db[0];
+    }
```

Comment:
Re: below comment, at least be consistent – either explicitly specify NULL, or don't :)

Reply:
Also, Heimdal style is generally to avoid unnecessary braces

Reply:
You could rewrite this as

```c
if (r->clientdb)
    hdb = r->clientdb;
else
    hub = config->db[0]
```

unless it was `config->db` you meant to check before dereferencing.

Reply:
Thanks.  I'll fix that.  Samba style is that the optional braces are a disaster waiting to happen so we always include them, so I do apologise for the style creep and thank you for your patience with us. https://gitlab.com/samba-team/samba/-/blob/master/README.Coding.md

Reply:
It isn't Heimdal style.  When in Rome...  We don't mind the extra braces, but don't expect us to write them.